Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandCharacterAutofixStuckInLocation.java

```diff
@@ -0,0 +1,70 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.CommonChecks;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.WebUtils;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+import com.universeprojects.miniup.server.commands.framework.Command.JavascriptResponse;
+
+/**
+ * Checks to see if a character is stuck in a location with no discoverable paths available. If the character
+ * is stuck, their locationKey is set to the parentLocationKey of their current location. 
+ * 
+ * @author papamarsh
+ * 
+ */
+public class CommandCharacterAutofixStuckInLocation extends Command {
+
+	public CommandCharacterAutofixStuckInLocation(ODPDBAccess db, HttpServletRequest request,
+			HttpServletResponse response) {
+		super(db, request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage {
+		ODPDBAccess db = ODPDBAccess.getInstance(request);
+		CachedDatastoreService ds = getDS();
+		
+		//grab the current character, its location, and the location's parent
+		CachedEntity character = db.getCurrentCharacter();
+		CachedEntity currentLocation = db.getEntity((Key) character.getProperty("locationKey"));
+		CachedEntity parentLocation = db.getParentLocation(ds, currentLocation);
+		
+		//special case for if user is in hell
+		if(currentLocation==db.getLocationByName("Ninth Circle of Hell"))
+		{
+			throw new UserErrorMessage("Nice Try! But there's no escape from Hell :)");
+		}
+		
+		//list the paths connected to the character's location
+		List<CachedEntity> pathList = db.getPathsByLocation(currentLocation.getKey());
+		for(CachedEntity checkPath:pathList)
+		{
+			CachedEntity checkPathLocation1 = db.getEntity((Key) checkPath.getProperty("location1Key"));
+			CachedEntity checkPathLocation2 = db.getEntity((Key) checkPath.getProperty("location1Key"));

```

Comment:
Btw you still need to fix the property name here.

As for the other thing... I see what you mean. I was just saying that we should wrap everything inside of the loop in a try-catch so that if anything is null in there (the only possibility at this point is that the path is null) then it would at least keep trying to find a valid path. I think we'll want to do more in this loop too. For example, if the character is in a combatsite, he shouldn't have to explore to find a path, it should be there already.  In that case, having the try-catch will be even more handy since the location could be null. It just makes it so you don't have to do null checks for everything. It's not the usual scenario of course, normally you don't want to swallow exceptions like that, but this very specific scenario is about trying to provide a workaround for unknown or known bugs, so it works.